nativelib/src/main/scala/scala/scalanative/unsafe/Word.scala

```diff
@@ -0,0 +1,353 @@
+package scala.scalanative
+package unsafe
+
+import scala.runtime.BoxesRunTime._
+import scala.reflect.ClassTag
+
+import scalanative.annotation.alwaysinline
+
+import scalanative.runtime._
+import scalanative.runtime.Intrinsics._
+import scalanative.runtime.Boxes._
+
+import scalanative.unsigned._
+
+final class Word(private[scalanative] val rawWord: RawWord) {
+  @alwaysinline def toByte: Byte   = castRawWordToInt(rawWord).toByte
+  @alwaysinline def toChar: Char   = castRawWordToInt(rawWord).toChar
+  @alwaysinline def toShort: Short = castRawWordToInt(rawWord).toShort
+  @alwaysinline def toInt: Int     = castRawWordToInt(rawWord)
+  @alwaysinline def toLong: Long   = castRawWordToLong(rawWord)
+
+  @alwaysinline def toUByte: UByte   = toByte.toUByte
+  @alwaysinline def toUShort: UShort = toShort.toUShort
+  @alwaysinline def toUInt: UInt     = toInt.toUInt
+  @alwaysinline def toULong: ULong   = toLong.toULong
+  @alwaysinline def toUWord: UWord   = new UWord(rawWord)
+
+  override def hashCode: Int = java.lang.Long.hashCode(toLong)
+
+  override def equals(other: Any): Boolean =
+    (this eq other.asInstanceOf[AnyRef]) || (other match {
+      case other: Word =>
+        other.rawWord == rawWord
+      case _ =>
+        false
+    })
+
+  override def toString(): String = toLong.toString
+
+  /**
+   * Returns the bitwise negation of this value.
+   * @example {{{
+   * ~5 == 4294967290
+   * // in binary: ~00000101 ==
+   * //             11111010
+   * }}}
+   */
+  @alwaysinline def unary_~ : Word =
+    (~toLong).toWord // TODO(shadaj): intrinsify
+
+  /** Returns the negated version of this value. */
+  @alwaysinline def unary_- : Word = 0 - this // TODO(shadaj): intrinsify
```

Comment:
Needs intrinsic.

Reply:
I'll skip commenting on any more `// TODO: intrinsify` comments, but all of them need to become intrinsics as well.